ranger-core/src/main/java/io/appform/ranger/core/model/ServiceRegistry.java

```diff
@@ -15,15 +15,22 @@
  */
 package io.appform.ranger.core.model;
 
+import java.util.concurrent.atomic.AtomicBoolean;
 import lombok.Data;
 
 import java.util.List;
 
 @Data
 public abstract class ServiceRegistry<T> {
     private final Service service;
+    private final AtomicBoolean initialRefreshCompleted = new AtomicBoolean(false);
 
     public abstract List<ServiceNode<T>> nodeList();
 
-    public abstract void updateNodes(List<ServiceNode<T>> nodes);
+    public abstract void update(List<ServiceNode<T>> nodes);
+
+    public void updateNodes(List<ServiceNode<T>> nodes) {
+        update(nodes);
+        initialRefreshCompleted.compareAndSet(false, true);
```

Comment:
This seems wrong. Initial refresh completed has a specific meaning. If you want to mean that at least one update has happened, please rename `initialRefreshCompleted` to `refreshed'. Also do not expose the atomic boolean directly as it is mutable. Create a different fucntion that responds back with a native boolean type.

Reply:
Make sense, made appropriate changes